codegen/model/chart.go

```diff
@@ -35,34 +35,44 @@ type Operator struct {
 
 	// these populate the generated ClusterRole for the operator
 	Rbac []rbacv1.PolicyRule
-	// these populate the generated ClusterRole for the operator
-	Volumes []v1.Volume
-	// mount these volumes to the operator container
-	VolumeMounts []v1.VolumeMount
-	// set these environment variables on the operator container
-	Env []v1.EnvVar
+
 	// add a manifest for each configmap
 	ConfigMaps []v1.ConfigMap
 
 	// if at least one port is defined, create a service for the
 	Service Service
-
-	// args for the container
-	Args []string
 }
 
 // values for Deployment template
 type Deployment struct {
 	// TODO support use of a DaemonSet instead of a Deployment
-	UseDaemonSet                bool
-	Image                       Image
-	Resources                   *v1.ResourceRequirements
+	UseDaemonSet bool
+	Container
+	Sidecars                    map[string]Container
```

Comment:
Just want to confirm that since this is going to be a breaking change regardless, we definitely want to have a separate "primary" container and sidecars versus a unified `[]Container`

CC: @ilackarms @EItanya